src/components/ScrollToTopButton.tsx

```diff
@@ -0,0 +1,30 @@
+import React, { useState, useEffect } from 'react'
+import '../styles/ScrollToTopButton.css'
+
+function ScrollToTopButton() {
+  const [isVisible, setIsVisible] = useState(false)
+
+  // Closing nav bar if user scrolls
+  useEffect(() => {
+    window.addEventListener('scroll', toggleVisible)
+    return () => window.removeEventListener('scroll', toggleVisible)
+  })
+
+  const toggleVisible = () => {
+    const scrolled = document.documentElement.scrollTop
+    if (scrolled > 150) setIsVisible(true)
+    else if (scrolled <= 0) setIsVisible(false)
```

Comment:
Nice! My suggestion here might be out of scope for this project but Isaac has made me too attentive to "further" simplifying statements like this. You don't have to make the changes but it is just good to know cuz we do it a lot in our codebase.
```suggestion
    setIsVisible((prev) => (scrolled > 150 && !prev) || prev)
```